hummingbot/connector/derivative/coinflex_perpetual/coinflex_perpetual_derivative.py

```diff
@@ -0,0 +1,1391 @@
+                    del self._account_positions[pos_key]
+                else:
+                    position.update_position(position_side=side,
+                                             unrealized_pnl=Decimal(position_data["positionPnl"]),
+                                             entry_price=Decimal(position_data["entryPrice"]),
+                                             amount=Decimal(amount))
+            else:
+                await self._update_positions()
+
+    async def _update_positions(self):
+        positions = await self._api_request(path=CONSTANTS.POSITION_INFORMATION_URL,
+                                            is_auth_required=True,
+                                            )
+        if not positions['data']:
+            return
+        for position in positions['data']:
+            trading_pair = position.get("instrumentId")
+            amount = Decimal(position.get("quantity"))
+            position_side = PositionSide.SHORT if amount < s_decimal_0 else PositionSide.LONG
+            unrealized_pnl = Decimal(position.get("positionPnl"))
+            entry_price = Decimal(position.get("entryPrice"))
+            leverage = Decimal('1')
+            pos_key = self.position_key(trading_pair, position_side)
+            if amount != 0:
+                self._account_positions[pos_key] = Position(
+                    trading_pair=await CoinflexPerpetualAPIOrderBookDataSource.convert_from_exchange_trading_pair(
+                        exchange_trading_pair=trading_pair,
+                        domain=self._domain,
+                        throttler=self._throttler,
+                        api_factory=self._api_factory,
+                    ),
+                    position_side=position_side,
+                    unrealized_pnl=unrealized_pnl,
+                    entry_price=entry_price,
+                    amount=amount,
+                    leverage=leverage
+                )
+            else:
+                if pos_key in self._account_positions:
+                    del self._account_positions[pos_key]
+
+    async def _fetch_order_status(self, tracked_order) -> Dict[str, Any]:
+        """
+        Helper function to fetch order status.
+        Returns a dictionary with the response.
+        """
+        order_params = {
+            "marketCode": await CoinflexPerpetualAPIOrderBookDataSource.convert_to_exchange_trading_pair(
+                hb_trading_pair=tracked_order.trading_pair,
+                domain=self._domain,
+                api_factory=self._api_factory,
+                throttler=self._throttler)
+        }
+
+        # If we get the exchange order id, use that, otherwise use client order id.
+        try:
+            async with timeout(self._sleep_time(1)):
+                await tracked_order.get_exchange_order_id()
+            order_params["orderId"] = tracked_order.exchange_order_id
+        except asyncio.TimeoutError:
```

Comment:
Since Coinflex allows order queries using the client order id we should not use the exchange order id and avoid waiting for it.

Reply:
See above regarding wait for order ID